main.js

```diff
@@ -1176,6 +1179,25 @@ function main() {
                     }
                 }
 
+                // CHeck setState counter per minute and stop script if too high
+                setStateCountCheckInterval = setInterval(() => {
+                    Object.keys(context.scripts).forEach(id => {
+                        const currentSetStatePerMinuteCounter = context.scripts[id].setStatePerMinuteCounter;
+                        context.scripts[id].setStatePerMinuteCounter = 0;
+                        if (currentSetStatePerMinuteCounter > adapter.config.maxSetStatePerMinute) {
+                            context.scripts[id].setStatePerMinuteProblemCounter++;
+                            adapter.log.debug(`Script ${id} has reached the maximum of ${adapter.config.maxSetStatePerMinute} setState calls per minute in ${context.scripts[id].setStatePerMinuteProblemCounter} consecutive minutes`);
+                            if (context.scripts[id].setStatePerMinuteProblemCounter > 1) {
```

Comment:
maybe put the threshold in a constant, this would be more self explanatory and less magic numbers +  a central place to change the threshold

Reply:
Because of the fact that this "2 mins in a row" thing is basically "prevent a script from being killed directly on start or such but allowing one minute" I would add kroe a comment. but I think it makes no sense to ever adjust that to a higher value